src/praisonai/praisonai/persistence/conversation/async_postgres.py

```
@@ -63,35 +63,41 @@ def __init__(
-        self._initialized = True
+        async with self._init_lock:
+            if self._initialized:
+                return
+            
+            try:
+                import asyncpg
+            except ImportError:
+                raise ImportError(
+                    "asyncpg is required for async PostgreSQL support. "
+                    "Install with: pip install asyncpg"
+                )
```

Ruff (0.15.9)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)